src/nupic/bindings/algorithms.i

```diff
@@ -101,9 +101,20 @@ _ALGORITHMS = _algorithms
 #include <nupic/proto/ConnectionsProto.capnp.h>
 #include <nupic/proto/SpatialPoolerProto.capnp.h>
 #include <nupic/proto/TemporalMemoryProto.capnp.h>
+%}
 
+//
+// Numpy API
+//
+%{
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
+%}
+%init %{
+  import_array();
```

Comment:
Good question about the `import_array` here: https://github.com/numenta/nupic.core/blob/942d993015912542c0f5f6ebaffd7e8f6a688f32/src/nupic/py_support/PyArray.cpp#L67

It's weird to do that repeatedly at runtime. That line should probably be moved to `engine_internal.i`.